### PR TITLE
Refactoring some traverses for performance reasons

### PR DESCRIPTION
**Pull request by TonioGela**

### CODE REVIEW

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Comment by TonioGela** [comment]
I'll wait for @antonioT90 or @iwoak to review the request before merging

**Review by antonioT90 — APPROVED** [review]

**Comment by antonioT90** [comment]
> I'll wait for @antonioT90 or @iwoak to review the request before merging

I've already seen it yesterday, i didn't approve cause you are the expert of this technology :D

**Comment by TonioGela** [comment]
> > I'll wait for @antonioT90 or @iwoak to review the request before merging
> 
> I've already seen it yesterday, i didn't approve cause you are the expert of this technology :D

;)

**Pull request merged**
